rust_src/src/coding.rs

```diff
@@ -0,0 +1,38 @@
+//! Coding system handler.
+
+use libc::c_int;
+
+use remacs_macros::lisp_fn;
+
+use crate::{
+    hashtable::HashLookupResult::{Found, Missing},
+    lisp::defsubr,
+    lisp::LispObject,
+    remacs_sys::{Fget, Qcoding_system_define_form, Vcoding_system_hash_table},
+};
+
+/// Return the ID of CODING_SYSTEM_SYMBOL.
+#[no_mangle]
+pub extern "C" fn CODING_SYSTEM_ID(object: LispObject) -> c_int {
```

Comment:
Is this going to be used in the C code? If not, you can drop the no_mangle and extern.

If this becomes a function only used by Rust, there is no need for the all caps either. This is not a C macro.

Reply:
`CODING_SYSTEM_ID` is used in C function which is not ported yet (e.g. `detect_coding_system` in `src/coding.c`).
I think we need no_mangle and extern for now until such functions are ported.

Do you think it is better to remain `CODING_SYSTEM_ID` being a C macro and define `fn coding_system_id(object: LispObject) -> isize` to use in Rust codes?

Reply:
Yes, that would be best I think.

Reply:
I see, thanks for your advice!